File: Data Structures & Algorithms/add-two-numbers/submission-6.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} l1
     * @param {ListNode} l2
     * @return {ListNode}
     */
    addTwoNumbers(l1, l2) {
        return this.add(l1, l2, 0)
    }

    add(l1, l2, carry) {
        if (!l1 && !l2 && carry === 0 ) {return null}
        
        let v1 = 0
        if (l1){
            v1 = l1.val
        }
        let v2 = 0
        if (l2){
            v2 = l2.val
        } 

        let newVal = v1 + v2
        if (carry > 0) {
            newVal++
            carry--
        }
        if (newVal > 9) {
            carry++
        }
        let newNode = new ListNode()
        newNode.val = newVal%10
        l1 = l1 ? l1.next : null
        l2 = l2 ? l2.next : null
        newNode.next = this.add(l1, l2, carry)
        return newNode
    }
}

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-6.go
/**
 * Definition for singly-linked list.
  * type ListNode struct {
   *     Val int
    *     Next *ListNode
	 * }
	  */

	  func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
	      if list1 == nil {return list2}
		  	if list2 == nil {return list1}

				var newHead *ListNode
					if list1.Val < list2.Val {
							tmp := list1.Next
									list1.Next = mergeTwoLists(tmp, list2)
											
													return list1
														} else {
																tmp := list2.Next
																		list2.Next = mergeTwoLists(list1, tmp) 
																				
																						return list2
																							}

																								return newHead
																								}
																								